### PR TITLE
James 1935 Optimize SetMessagesCreatorProcessor by reduce getMailboxes method

### PR DESCRIPTION
**Pull request by quynhn**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Role.java [hunk 1]
```diff
@@ -34,14 +36,14 @@
...
-    public static final Role SPAM = new Role("spam");
-    public static final Role TEMPLATES = new Role("templates");
+    public static final Role INBOX = new Role(DefaultMailboxes.INBOX);
+    public static final Role DRAFTS = new Role(DefaultMailboxes.DRAFTS);
+    public static final Role OUTBOX = new Role(DefaultMailboxes.OUTBOX);
+    public static final Role SENT = new Role(DefaultMailboxes.SENT);
+    public static final Role TRASH = new Role(DefaultMailboxes.TRASH);
+    public static final Role ARCHIVE = new Role("Archive");
+    public static final Role SPAM = new Role("Spam");
+    public static final Role TEMPLATES = new Role("Templates");
```
Comment (rouazana) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Role.java]: Unfortunately these roles are defined in the JMAP specification, you cannot change them like this.
http://jmap.io/spec-mail.html#mailboxes

Comment (quynhn) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Role.java]: Then I rolled it back.
And I changed a bit on code with searching by expression. I assume that we will search ignore case sensitive such as mailbox's role was there so it does not make sense getting all mailbox then filter by name.
For further, if we need to support search with case sensitive than we can add more parameter on searching method.
WDYT?

Comment (chibenwa) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/model/mailbox/Role.java]: What would you think of mapping a Role to a default mailbox, then we can ensure the value of the default mailbox for case, and default to search, without breaking anything ?

**Comment by aduprat** [comment]
test this please

**Comment by aduprat** [comment]
replaced by #624

**Pull request closed without merging**